src/main/java/jenkins/plugins/git/GitToolChooser.java

```diff
@@ -209,13 +172,48 @@ private GitTool resolveGitToolForRecommendation(GitTool userChoice, String recom
                 return null;
             }
         } else {
```

Comment:
This branch of logic is designed for repo size > 5M. 
Here two things can happen: 
- if user's choice is not jgit or jgitapache, it would be a command line git tool. We would return that same choice without any concern.
- if the user's choice is jgit or jgitapache, it would look for the right command line git tool using `recommendGitToolOnAgent`.